telemetry/specs/INT.mdk

```diff
@@ -384,44 +384,71 @@ Considering all these factors, the INT specification requires that an INT
 switch **must not fragment** packets in order to append INT information to
 the packet.
 
-### Checksum Update
+## Checksum Update
 
 INT may be transported over an L4 protocol such as TCP or UDP, or over an
 encapsulation header that includes an L4 header, such as VXLAN. In these cases,
 an INT source or transit switch may be required to update the L4 Checksum of
 the encapsulating header.
 
-In some cases, an L4 Checksum update is not necessary. For example, when UDP is 
-transported over IPv4, it is possible to assign a zero Checksum, causing the 
-receiver to ignore the value of the Checksum field. For UDP over IPv6, there
-are specific use cases in which it is possible to assign a zero Checksum 
-(as defined in RFC 6936).
-
-If an L4 Checksum update is required, an INT source/transit switch may perform
-it in one of two ways:
-
-* Update the L4 Checksum field such that the new value is equal to the Checksum 
+In some cases, an Layer 4 Checksum update is not necessary. For example,
+encapsulation protocols such as VXLAN and Geneve allow for bypassing validation
+of outer Layer 4 checksum at the receiving tunnel endpoint, since the integrity
+of the encapsulated payload is covered by the inner Layer 4 checksum. VXLAN
+and Geneve use UDP as the Layer 4 protocol and populate the outer Layer 4
+checksum with zero to indicate that this checksum value is to be ignored as
+permitted by RFC 768 and RFC 6936. However, it is possible for certain
+implementations of TCP-based encapsulation protocols such as Stateless
+Transport Tunneling Protocol (STT) to also ignore outer Layer 4 checksum at the
+receiving tunnel endpoint based on the same rationale that the integrity
+of the encapsulated payload is covered by the inner Layer 4 checksum.
+Another scenario where Layer 4 checksum validation may be bypassed is when INT is
+used to fetch network state via probe packets, not real data packets.
+
+This specification defines a bit in the INT header that may be set at the INT
+source to indicate to transit INT switches that outer Layer 4 checksum
+update is not required as they insert INT metadata in the packet. This bit may
+be set with caution at an INT source that is enabling INT on probe packets,
+or at an INT source that is also a Tunnel Endpoint and knows that the
+receiving Tunnel Endpoint does not validate outer Layer 4 checksum.
+
+If an INT transit hop receives UDP packets with checksum equal to zero,
+or a packet with this "Checksum update optional" bit set in the INT header,
+it may choose to not update the Layer 4 checksum upon inserting INT metadata.
+If an INT transit hop does update the Layer 4 checksum in such packets, it
+**must** perform a full checksum calculation as the original checksum in the
+packet cannot be assumed to be valid for an incremental checksum calculation.
```

Comment:
I see two different cases:

1. If it receives UDP packets with checksum equal to zero, then I don't think the INT spec is the place to define how to change the checksum. IETF RFCs specify how to behave when receiving a packet with UDP checksum equal to zero. IMO this is out of scope of INT and does not need to be discussed in this spec.

2. If the "checksum update optional" bit is set, then it does not matter what the INT transit hop does. The resulting checksum must be ignored at the receiving tunnel endpoint. There is no need for this requirement.